### PR TITLE
coreos-boot-edit: abort if system has several boot partitions

### PR DESCRIPTION
**Pull request by nikita-dubrovskii**
This requires https://github.com/coreos/coreos-installer/pull/658

Signed-off-by: Nikita Dubrovskii <nikita@linux.ibm.com>

### CODE REVIEW

File: overlay.d/05core/usr/lib/dracut/modules.d/35coreos-ignition/coreos-boot-edit.sh [hunk 1]
```diff
@@ -14,6 +14,9 @@ karg() {
     echo "${value}"
 }
 
+# Ensure we have a unique filesystem labeled 'boot'
+/usr/bin/rdcore verify-unique-fs-label boot
```
Comment (bgilbert) [inline overlay.d/05core/usr/lib/dracut/modules.d/35coreos-ignition/coreos-boot-edit.sh]: For consistency with the rest of this file, let's not use an absolute path.

**Review by bgilbert — DISMISSED** [review]
Looks good; just waiting for a coreos-installer release.

**Comment by jlebon** [comment]
I think we'll also want this somewhere at the beginning of the initramfs too as per https://github.com/coreos/fedora-coreos-tracker/issues/976#issuecomment-953147348, but we can do that in a follow-up.

**Review by bgilbert — DISMISSED** [review]

**Comment by jlebon** [comment]
I think we still need the check in `coreos-boot-edit` (which is the one that will take the new `--rereadpt` switch from https://github.com/coreos/coreos-installer/pull/731). Or did you mean to do that in a separate PR?

File: overlay.d/05core/usr/lib/dracut/modules.d/35coreos-ignition/coreos-unique-boot.service [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+OnFailure=emergency.target
+OnFailureJobMode=isolate
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStart=/usr/bin/rdcore verify-unique-fs-label boot
+
+
```
Comment (jlebon) [inline overlay.d/05core/usr/lib/dracut/modules.d/35coreos-ignition/coreos-unique-boot.service]: Extraneous blank lines

**Review by jlebon — DISMISSED** [review]
LGTM! Need to hold until coreos-installer release with https://github.com/coreos/coreos-installer/pull/731.

Would be good to have tests that check the error conditions, though this would require some work in kola. How did you test this? A few checks I can think of not covered by CI:
- Verifying that `coreos-unique-boot.service` errors out if we boot with two bootable disks (or really just two disks each with a filesystem labeled `boot`).
- Verifying that `coreos-boot-edit.service` errors out. That one might be trickier to check. Hmm, actually it'd work to have an Ignition config which creates a filesystem labeled `/boot` on some secondary disk.

**Comment by nikita-dubrovskii** [comment]
> Would be good to have tests that check the error conditions, though this would require some work in kola. How did you test this? 

I did it manually (on s390x):
 - attaching 2nd `boot` disk
 - with `multipath` 
 - `multipath` +  `boot` disk

**Comment by jlebon** [comment]
We have new coreos-installer now. Restarted CI on this!

**Review by jlebon — APPROVED** [review]

**Comment by bgilbert** [comment]
The `multipath.day1` kola test has started flaking; might be relevant to this PR: https://github.com/coreos/fedora-coreos-tracker/issues/1105

**Pull request merged**